tasks/InfracostComment/index.ts

```diff
@@ -92,31 +92,66 @@ function formatOutput(output: string, targetType: string, behavior: string): str
   return output;
 }
 
+interface RepoProvider {
+  provider: string,
+  inputToken: string,
+  envVariable: string,
+  commentFormat: string,
+}
+
+const providerOptions: RepoProvider[] = [
+  {
+    provider: 'GitHub',
+    inputToken: 'githubToken',
+    envVariable: 'GITHUB_TOKEN',
+    commentFormat: 'github-comment',
+  },
+  {
+    provider: 'TfsGit',
+    inputToken: 'accessToken',
+    envVariable: 'SYSTEM_ACCESSTOKEN',
+    commentFormat: 'azure-repos-comment',
+  },
+]
+
+function setupRepoProvider(): RepoProvider {
+  const env = taskLib.getVariable('BUILD_REPOSITORY_PROVIDER');
+  const provider = providerOptions.find(opts => opts.provider === env);
```

Comment:
this is nice magic - but would prefer just to use a plain `map/object` instead of array so we can just do a lookup by key instead of having to loop over.

Reply:
e.g.

```ts
const provider = providerOptions[env];
```